data_driven_onboarding/tests/test_schema.py

```diff
@@ -56,6 +56,15 @@ def test_it_works(self, settings, requests_mock, monkeypatch):
             "unitCount": 123,
         }
 
+    def test_it_returns_none_when_bbl_is_not_in_database(self, settings, monkeypatch):
+        settings.WOW_DATABASE = "blah"
+        monkeypatch.setattr(
+            schema,
+            "run_ddo_sql_query",
+            lambda bbl: [],
+        )
+        assert self.request("address where bbl is missing from db", "Brooklyn") is None
```

Comment:
Hmm, this is the one blocker--it looks like the monkeypatch function you're providing isn't actually being called.  The following code uses Python's [`MagicMock`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) to make sure that it's called:

```diff
     def test_it_returns_none_when_bbl_is_not_in_database(self, settings, monkeypatch):
         settings.WOW_DATABASE = "blah"
+        from unittest.mock import MagicMock
+
+        fake_run_ddo_sql_query = MagicMock(return_value=[])
         monkeypatch.setattr(
             schema,
             "run_ddo_sql_query",
-            lambda bbl: [],
+            fake_run_ddo_sql_query,
         )
         assert self.request("address where bbl is missing from db", "Brooklyn") is None
+        fake_run_ddo_sql_query.assert_called_once()
```

But with this code added, the test fails 😣

I should add that I actually only noticed this because the [Chrome CodeClimate extension](https://codeclimate.com/browser-extension/) also shows code coverage!  Here's what it looks like when I'm reading the PR:

> ![image](https://user-images.githubusercontent.com/124687/115044252-53ef9480-9ea3-11eb-8e05-8f95396a0ebf.png)

The dark green bar on the left (next to the line number) indicates that the line was covered by a test, while the orange bar indicates it wasn't.  When I noticed all the `return None` lines had an orange bar next to them, I thought something funky might be happening!

Reply:
Aha!! Two things:
1. OK, this is kind of what I thought - before, I was calling `requests_mock.get(settings.GEOCODING_SEARCH_URL, json=EXAMPLE_SEARCH)` in the new test as well, but it would only return the monkeypatch values from the `test_it_works` test, not the new one. I'm theorizing that even tho monkeypatch is supposed to reset all the values from `setattr` at the end of a test, something we're using is preventing it from doing that (and actually preventing the variable from being reset in the new test). I tried changing the order of the tests and it seems whichever one calls `setattr` first sticks. Let me get this back to what I had before and maybe we can chat about it on a quick call.

2. I installed the codeclimate extension, but somehow couldn't get it to recognize justfix repos. help on this would be much appreciated! I'm also eager to get codeclimate in our devcontainer - this extension has some instructions. monovaldes.linter-codeclimate

Reply:
Weirddd, I added back requests mock and the monkeypatch problem was gone. MagicMock says monkeypatch is clearly getting called, and it's fine with resetting the same variable from a previous test. Pushing now to see if codeclimate agrees (until i can get it up and running on my vscode)

Reply:
OK, I've gotten codeclimate working with the justfix repos, and it says the code is now being hit by the tests. Pretty sure this solves it.

<img width="663" alt="Screen Shot 2021-04-16 at 2 22 15 PM" src="https://user-images.githubusercontent.com/1595717/115067716-27894780-9ea6-11eb-9d38-dfb91513a282.png">

Reply:
oooo nice!!  you should show us how you set this up at our next dev pow wow, i'd like to get it set up on my thingy too.  and let's see if we can somehow reproduce what you were running into with the monkeypatch thing not resetting between tests... as you intuited, monkeypatch _should_ reset everything between tests but if that's not happening, we should definitely look into it!